primenet.py

```diff
@@ -5333,18 +5333,19 @@ def report_result(adapter, adir, sendline, ar, tasks, retry_count=0):
     elif result_type in {PRIMENET.AR_TF_FACTOR, PRIMENET.AR_TF_NOFACTOR}:
         args["d"] = (
             1
-            if ar["rangecomplete"]
-            and not any(
+            if not any(
                 task.k == assignment.k and task.b == assignment.b and task.n == assignment.n and task.c == assignment.c
                 for task in tasks
                 if isinstance(task, Assignment)
             )
             else 0
         )
         args["sf"] = ar["bitlo"]
+        if ar["rangecomplete"]:
+            args["ef"] = ar["bithi"]
```

Comment:
I think we want to include ef in the found factor API call (I just got a found factor that finished the range, but was reported as a "bald" factor because there was no ef in the API call in the previous iteration of the code)

In the case that StopAfterFactor is in [0,1] in mfaktc.ini, this will be represented in the results.json.txt as rangecomplete=True, StopAfterFactor=2 will be represented as rangecomplete=False

So we'll use rangecomplete to determine whether or not the range has been "finished", and only the existence of more assignments for the exponent to determine whether we're done with the AID.

Reply:
Sounds good. The changes I previously made were just based off of what Prime95/MPrime does, but mfaktc of course has more advanced TF features, so it makes sense that you might need to do things differently.

Specifically, Prime95/MPrime reports each factor as soon as it is found, so it then makes a separate `NO_RESULT` API call after completing the range to mark the assignment as done.

Reply:
There may have been a nugget of truth to the way you had implemented it, I reached out to James about the fact that I was still seeing my reported factors as "bald factors" but it was really just a visual bug due to a sql pretty printing function failing to account for json ordering, and not printing the range the factor was found in. The underlying metadata was still the same.

James said primenet just deletes the exponent from the "factor depth table" if there's any factor, so it might not make a difference either way (with regards to `ef=`), and as for the "multiplying factors in the same range together to report them all" bit, I'll get clarification on that and alter this script accordingly as well.

(We at least know that the API can handle numbers larger than uint64, as I just fed it a 72 bit factor, so I assume it's arbitrary precision and could handle composite factors)

Reply:
> We at least know that the API can handle numbers larger than uint64, as I just fed it a 72 bit factor, so I assume it's arbitrary precision and could handle composite factors

In section 5.7.5.1.1 the PrimeNet v5 API documentation says there is a "length limit [of] 512 digits", while section 8.2 it says "max length 200". Hexadecimal is also allegedly supported. I would presume the limit is actually larger than both of these numbers, but probably not arbitrary.